changelog/unreleased/issue-2571

```diff
@@ -0,0 +1,14 @@
+Enhancement: Self-heal missing file parts during backup of unchanged files
+
+We've improved the resilience of restic to certain types of repository damages.
+For files that are unchanged since the parent snapshot, the backup command now
+verifies that all file parts still exist in the repository.
+Parts that are missing from a damaged repository are backed up again.
+That verification was already run for files that were modified since the parent snapshot.
+
+Note that restic still will not backup the contents if they are contained in the index
+but the actual data is not present. This situation can only be detected by restic check.
+Please ensure that you run `restic check` regularly.
```

Comment:
Overall I think this looks good, but I'd like to make the following changes (writing a new copy of the entire two paragraphs for simplicity - please copy all of it and replace the old text with it):

~~~
We've improved the resilience of restic to certain types of repository corruption.

For files that are unchanged since the parent snapshot, the backup command
now verifies that all parts of the files still exist in the repository. Parts that are
missing, e.g. from a damaged repository, are backed up again. This verification
was already run for files that were modified since the parent snapshot, but is
now also done for unchanged files.

Note that restic will not backup file parts that are referenced in the index but
where the actual data is not present on disk, as this situation can only be
detected by restic check. Please ensure that you run `restic check` regularly.
~~~

Reply:
is changed.